bin/pkg/lock.ml

```diff
@@ -94,6 +95,7 @@ let solve_lock_dir
       ~unset_solver_vars_from_context:
         (unset_solver_vars_of_workspace workspace ~lock_dir_path)
   in
+  let time_start = Unix.gettimeofday () in
```

Comment:
The `Unix` documentation is very light on this but usually `[gettimeofday](https://www.man7.org/linux/man-pages/man2/gettimeofday.2.html)` is not a good way to determine the runtime because it is not monotonic, thus affected by time changing e.g. on DST or system clock adjustments.

Looks like `mtime` exists exactly for this.

Reply:
Good point, but mtime isn't exposed in the `Unix` module and there's already a precedent for using `gettimeofday` for measuring durations of time in dune.

Reply:
Yeah, I also noticed that despite what the documentation of mtime says, to actually get the current mtime you need the C/JS stub, so there's no easy way to get that without vendoring another library.